dataconnect/app/src/pages/Movie.tsx

```diff
@@ -0,0 +1,276 @@
+import { useContext, useEffect, useState } from 'react';
+import { useParams, Link } from 'react-router-dom';
+import { MdFavorite, MdFavoriteBorder, MdStar } from 'react-icons/md';
+import {
+  getMovieById,
+  GetMovieByIdResponse,
+  addFavoritedMovie,
+  deleteFavoritedMovie,
+  getIfFavoritedMovie,
+  addReview,
+  deleteReview,
+  searchMovieDescriptionUsingL2similarity,
+  SearchMovieDescriptionUsingL2similarityResponse,
+} from '@movie/dataconnect';
+import { onAuthStateChanged, User } from 'firebase/auth';
+import { AuthContext } from '@/lib/firebase';
+import NotFound from './NotFound';
+
+export default function MoviePage() {
+  const { id } = useParams() as { id: string };
+  const [movie, setMovie] = useState<GetMovieByIdResponse['movie'] | null>(null);
+  const [loading, setLoading] = useState(true);
+  const [authUser, setAuthUser] = useState<User | null>(null);
+  const [isFavorited, setIsFavorited] = useState(false);
+  const [reviewText, setReviewText] = useState('');
+  const [userReview, setUserReview] = useState<GetMovieByIdResponse['movie']['reviews'][0] | null>(null);
+  const [rating, setRating] = useState(0);
+  const [similarMovies, setSimilarMovies] = useState<SearchMovieDescriptionUsingL2similarityResponse['movies_descriptionEmbedding_similarity']>([]);
+  const auth = useContext(AuthContext);
+
+  useEffect(() => {
+    const unsubscribe = onAuthStateChanged(auth, (user) => {
+      async function checkIfFavorited() {
```

Comment:
I noticed seeing this twice, is it worth extracting out to a util function or do you think it'll be less clear that way?

Reply:
This was to avoid using React Suspense while still be able to call an async function in useEffect... as an util function, I think it would still need to be wrapped in separate useCallbacks?